### ISSUE TITLE
crossEntropy still validates its targets below the backend dispatch

### ISSUE THREAD

**Issue by felladrin** [felladrin/gguf-trainer#82]
Noted in lever 30 while fixing #61, and the last instance of the pattern those PRs argued against.

`crossEntropy` dispatches to the backend before it validates:

```ts
export function crossEntropy(logits: Tensor, targets: number[]): Tensor {
  assertMatrix(logits, "logits", "crossEntropy");
  if (opsBackend) return opsBackend.crossEntropy(logits, targets);
  const [T, V] = logits.shape;
  const kept = keptRowsInVocab(targets, T, V, "crossEntropy");
```

The rank check is above the dispatch, from #72. The id check is not: `keptRowsInVocab` runs in the CPU body, and the GPU path is covered only because `webgpu.ts` calls it separately. Same for `fusedCrossEntropy`, which calls it in both places.

There is no live gap. All four call sites are present, and `targetRangeGate` in `tests/gpu-parity.ts` pins both device ones. But this is exactly the shape that produced #61: validate under the dispatch and every implementation needs its own call, and the one nobody remembers is the one that ships unvalidated. `embedding` (#69) and `softCrossEntropy` (#70) were both moved above the dispatch for that reason, and this is the one that was left.

The reason it was left is a real cost, not an oversight. `keptRowsInVocab` returns the kept-row count, and both backends need it as their loss denominator. Hoisting the call means either:

- the GPU path recomputes the count, an extra O(T) pass it does not have today; or
- `OpsBackend.crossEntropy` and `.fusedCrossEntropy` take `kept` as an argument, which changes the interface for a guard.

Neither is obviously right, which is why this is a question rather than a patch. The third option is to leave it and accept that two of the five validators sit below the dispatch, with a comment saying so and why. That is defensible, and it is what the code does now minus the comment.

Whatever lands, `targetRangeGate` already has the arm that would catch a regression: moving a guard below the dispatch passes every CPU case and fails only there.
